console/client/src/providers/tabs-provider.tsx

```diff
@@ -12,23 +12,65 @@ export type Tab = {
 }
 
 export const TabSearchParams = {
-  active: 'active-tab',
+  id: 'tab-id',
+  type: 'tab-type',
 } as const
 
 export const timelineTab = { id: 'timeline', label: 'Timeline', type: TabType.Timeline }
 
+export type ActiveTab = {
+  id: string;
+  type: string;
+} | undefined
+
 type TabsContextType = {
   tabs: Tab[];
-  activeTab?: number;
+  activeTab?: ActiveTab;
   setTabs: React.Dispatch<React.SetStateAction<Tab[]>>;
-  setActiveTab: React.Dispatch<React.SetStateAction<number>>;
+  setActiveTab: React.Dispatch<React.SetStateAction<ActiveTab>>;
 }
 
-export const TabsContext = React.createContext<TabsContextType>({ tabs: [], activeTab: 0, setTabs: () => { }, setActiveTab: () => { } })
+export const TabsContext = React.createContext<TabsContextType>({
+  tabs: [],
+  activeTab: undefined,
+  setTabs: () => { },
+  setActiveTab:  () => {},
+})
+
+export const isValidTab = ({ id, type }: {id?: string; type?: string}): boolean => {
+  //P1 no ID or type
+  if(!id || !type) {
+    // throw new Error(`required tab field undefined: ${JSON.stringify({ type, id })}`)
+    return false
```

Comment:
For these, should we throw or `return false`? Or are these comments here for future work?

Reply:
Not anymore as I stated in my comment above I need to add a error message componen we talked about this in stand up